docs/resources/storage_bucket.md

```diff
@@ -0,0 +1,75 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "boundary_storage_bucket Resource - terraform-provider-boundary"
+subcategory: ""
+description: |-
+  The storage bucket resource allows you to configure a Boundary storage bucket. A storage bucket can only belong to the Global scope or an Org scope. At this time, the only supported storage for storage buckets is AWS S3. This feature requires Boundary Enterprise or Boundary HCP.
+---
+
+# boundary_storage_bucket (Resource)
+
+The storage bucket resource allows you to configure a Boundary storage bucket. A storage bucket can only belong to the Global scope or an Org scope. At this time, the only supported storage for storage buckets is AWS S3. This feature requires Boundary Enterprise or Boundary HCP.
+
+## Example Usage
+
+```terraform
+resource "boundary_scope" "org" {
+  name                     = "organization_one"
+  description              = "My first scope!"
+  scope_id                 = boundary_scope.global.id
+  auto_create_admin_role   = true
+  auto_create_default_role = true
+}
+
+resource "boundary_storage_bucket" "aws_example" {
+  name            = "My aws catalog"
+  description     = "My first host catalog!"
+  scope_id        = boundary_scope.org.id
+  plugin_name     = "aws"
+  bucket_name     = "mybucket"
+  attributes_json = jsonencode({ "region" = "us-east-1" })
+
+  # recommended to pass in aws secrets using a file() or using environment variables
+  # the secrets below must be generated in aws by creating a aws iam user with programmatic access
+  secrets_json = jsonencode({
+    "access_key_id"     = "aws_access_key_id_value",
+    "secret_access_key" = "aws_secret_access_key_value"
+  })
+  worker_filter = "\"pki\" in \"/tags/type\""
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `attributes_json` (String) The attributes for the storage bucket. Either values encoded with the "jsonencode" function, pre-escaped JSON string, or a file:// or env:// path. Set to a string "null" or remove the block to clear all attributes in the storage bucket.
```

Comment:
Can we clarify that the region attribute is the required attribute?

Reply:
All the other required fields have `storagebucket` options. For example: [bucketName](https://github.com/hashicorp/boundary/blob/7f1d20ff3a95778bce75be8b02c4544bb1fb1995/api/storagebuckets/option.gen.go#L107).
They are all eventually set as storage bucket attributes in Boundary. Do you know why we didn't follow the same pattern with `region` field?

Reply:
Maybe a question for @louisruch or @ddebko who worked on storage 👆

Reply:
Region is technically not required, it is only required for aws. While bucket name will required for all plugins.

Reply:
@louisruch as far as documenting this in the provider- Should we keep `attributes_json` as an optional field but add a note that for AWS the region attribute is required?

Reply:
Yeah it makes sense for now to explicitly call this out. Though once we have multiple plugins it might get a bit verbose to list all required attributes per plugin

Reply:
So Boundary will rather handle validation of the `region`. I can keep it optional & update the docs saying it's required when using the AWS plugin

I also looked into `TF RequiredWith` && [TF Schema ValidateFunc](https://developer.hashicorp.com/terraform/plugin/sdkv2/schemas/schema-behaviors#validatefunc) to use the value of another field to validate if `attributes_json ` should be required but that does not seem possible because I can't read the value of other fields from `ValidateFunc`